tools/clang/lib/CodeGen/CGHLSLMS.cpp

```diff
@@ -985,6 +1001,12 @@ unsigned CGMSHLSLRuntime::ConstructStructAnnotation(DxilStructAnnotation *annota
           ConstructFieldAttributedAnnotation(fieldAnnotation, fieldTy,
                                              bDefaultRowMajor);
           fieldAnnotation.SetCBufferOffset(CBufferOffset);
+          if (StartBitOffset > 0) {
+            DxilFieldAnnotation &prevFieldAnnotation =
+                annotation->GetFieldAnnotation(fieldIdx - 1);
```

Comment:
Can `fieldIdx` be zero? What if the struct is just one member with a bitfield width?

Reply:
If fieldIdx is zero then, StartBitOffset will be zero too.